src/app/backend/handler/localehandler.go

```diff
@@ -131,46 +122,45 @@ func (handler *LocaleHandler) determineLocalizedDir(locale string) string {
 	locales := handler.SupportedLocales
 	tag, _, confidence := language.NewMatcher(locales).Match(tags...)
 
-	if confidence < language.Exact {
-		tag, confidence, err = mapLocale(locale, locales)
-		if err != nil {
-			return defaultDir
-		}
-	}
-
-	matchedLocale := tag.String()
-	// If locale match is exact, then we have to manually look for proper locale code as language
-	// library contains a bug that returns invalid locale string.
-	// Related issue: https://github.com/golang/go/issues/24211
-	if confidence == language.Exact {
-		matchedLocale = ""
-		for _, l := range locales {
-			base, _ := tag.Base()
-			if l.String() == base.String() {
-				matchedLocale = l.String()
+	localeDir := ""
+	switch confidence {
+	case language.Exact:
+		// If locale match is exact, then we have to manually look for proper locale code as language
+		// library contains a bug that returns invalid locale string.
+		// Related issue: https://github.com/golang/go/issues/24211
+		// TODO(shu-mutou): Handling with `Extensions` is hack for the bug.
+		// So possively we need to fix again this.
+		// More details, see https://github.com/kubernetes/dashboard/issues/4973
+		if len(tag.Extensions()) > 0 {
+			// e.g. map `fr-FR` to `fr`, `zh-CN` to `zh`, `zh-TW` to `zh-Hant`
+			localeDir = getSupportedLocaleDir(tag.Parent().String(), locales)
```

Comment:
I wouldn't want to rely on their internal implementation here and map specific regional locale to more generic ones. We do know already that they have an issue with returning proper locales. Relying too much on their internal logic might result in further issues. Let's stick to the basic check and rely only on confidence. If it's exact, then look for proper locale on our own as it was done before.

Reply:
I found the issue is not simply bug in `golang.org/x/text/language`. If we can set supported locales properly, it matches locales properly. Matcher maps `zh-SG` and `zh-HK` to `zh-Hans-SG` and `zh-Hant-HK`. And now, in Angular>=9.0.5 supports `zh-Hans-SG` and `zh-Hant-HK`, so we can build with them and we do not need to replace them. However, we need to replace `zh-CN` and `zh-TW` yet.

Reply:
It is a bug if it returns weird `xx-yy-zzzzzzz-n` strings.

Reply:
After all, it seems weird but does not seem to be a bug. See more,  https://godoc.org/golang.org/x/text/language#NewMatcher